### ISSUE TITLE
Bug: Custom pipeline templates cannot be initialized from git repositories

### ISSUE THREAD

**Issue by kfr2** [aws/aws-sam-cli#4206]
### Description:
Whenever a git repository is passed to the "Custom Pipeline Template Location" option of `sam pipeline init`, the sam process exits with the error message "Error: questions definition file not found at [...]/questions.json" after cloning the repository. This error state was introduced by https://github.com/aws/aws-sam-cli/pull/4108 by calling `self._generate_from_pipeline_template()` outside the `osutils.mkdir_temp()` context manager. 

### Steps to reproduce:
```
❯ sam pipeline init

sam pipeline init generates a pipeline configuration file that your CI/CD system
can use to deploy serverless applications using AWS SAM.
We will guide you through the process to bootstrap resources for each stage,
then walk through the details necessary for creating the pipeline config file.

Please ensure you are in the root folder of your SAM application before you begin.

Select a pipeline template to get started:
	1 - AWS Quick Start Pipeline Templates
	2 - Custom Pipeline Template Location
Choice: 2
Template Git location: git@github.com:aws/aws-sam-cli-pipeline-init-templates.git

Cloning from git@github.com:aws/aws-sam-cli-pipeline-init-templates.git (process may take a moment)
Error: questions definition file not found at /var/folders/h2/8fby63l10_13mjqghqtl4pzw0000gq/T/tmp9opdbn_l/custom-pipeline-template/questions.json
```

### Expected result:
sam should be able to clone the repository and render the templates contained therein.

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)
1. OS: Mac OS -- but this is a platform-agnostic issue.
2. `sam --version`: 1.56.0
3. AWS region: N/A

**Comment by lucashuy** [comment]
Thanks for raising this issue and the PR fix for the team to review! Any suggestions we have to your fix will be provided on the PR

**Comment by hawflau** [comment]
Fix is released in SAM CLI v1.60.0. Closing.
